src/Components/AddTask/AddTask.tsx

```diff
@@ -1,67 +1,114 @@
 
 import axios from "axios";
 import { useContext, useEffect, useState } from "react";
-import { useNavigate } from "react-router-dom";
+import { Link, useNavigate } from "react-router-dom";
 import { AuthContext } from "./../../Context/AuthContext";
 import { ToastContext } from "../../Context/ToastContext";
 import { SubmitHandler, useForm } from "react-hook-form";
 import { ProjectContext } from '../../Context/ProjectContext';
-import { FormValues, projectType ,IAuth} from './../../Types/Types';
+import { FormValues, projectType, IAuth } from './../../Types/Types';
 
 
 const AddTask: React.FC = () => {
-  const { baseUrl, requestHeaders }:any
-  //Pick<IAuth, 'baseUrl','requestHeaders'> 
-  = useContext(AuthContext);
+  const { baseUrl, requestHeaders }: any
+    //Pick<IAuth, 'baseUrl','requestHeaders'>
+    = useContext(AuthContext);
   const { getToastValue }: void = useContext(ToastContext);
-  const {  projects }: projectType[] = useContext(ProjectContext);
+  // const { projects }: projectType[] = useContext(ProjectContext);
   // console.log(projects, "from addtasks");
   const [isLoading, setIsLoading] = useState(false);
   const navigate = useNavigate();
+  const [userList, setUserList] = useState([])
+  const [projectList,setProjectList]=useState([])
+
 
-  
   const {
     register,
     handleSubmit,
     formState: { errors },
   } = useForm<FormValues>();
 
-  const goBack=()=>{
+  const goBack = () => {
     navigate("/dashboard/tasks");
   }
   const onSubmit: SubmitHandler<FormValues> = async (data: FormValues) => {
     // setIsLoading(true)
     console.log(data);
-    // axios
-    //   .post(`${baseUrl}/Task`, data, { headers: requestHeaders })
-    //   .then((response) => {
-    //     setIsLoading(false)
-    //     getToastValue(
-    //       "success",
-    //       response?.data?.message || "Project added suceessfully"
-    //     );
-    //     navigate("/dashboard/projects");
-    //   })
-    //   .catch((error) => {
-    //     setIsLoading(false)
-    //     getToastValue(
-    //       "error",
-    //       error?.response?.data?.message ||
-    //         "An error occurred. Please try again."
-    //     );
-    //   });
+    axios
+      .post(`${baseUrl}/Task`, data, { headers: requestHeaders })
+      .then((response) => {
+        setIsLoading(false)
```

Comment:
Consider using finally(() => setIsLoading(false)) after the promise to consistently reset the loading state to false, avoiding redundancy in both the then and catch blocks.